### PR TITLE
Refactor prepare add

### LINKED ISSUE #156: Code quality feedback for A0139257X
# main
- too many very long method signatures, please resolve them
- many methods missing header comments
# test
- you can move much of the setup to independent methods and just call it (or better still use `@Before`) in each test; this eliminates code duplication that is present now
- too many "long lines of code"; split them appropriately to improve readability
- test names do not follow the naming convention

Details [here](https://github.com/nus-cs2103-AY1617S1/addressbook-level4/pull/94#pullrequestreview-6360514)

### LINKED ISSUE #166: refactor prepareAdd method
Came across your `prepareAdd` method while on a random quality check.
Please refactor to improve SLAP. 
NEVER write methods that long :-p

### LINKED ISSUE #222: [Add Command] Bug caused dates before today being added and became a week later

### PR DESCRIPTION
**Pull request by marcusngwenjian**
This closes #156 , closes #166 , closes #222

@CS2103AUG2016-W15-C3/testing Ready for testing
@CS2103AUG2016-W15-C3/documentation Ready for documentation

### CODE REVIEW

**Comment by ZuoZhuolin** [comment]
Test reviewed

**Comment by turtle96** [comment]
1. add smth on 10-11
   image
2. add hello on 10-nov
   image

2nd command grabs the date, it would be good if 1st command can grab the date as well

**Comment by turtle96** [comment]
Testing pass

**Comment by marcusngwenjian** [comment]
@turtle96 
No, The first should not grab the date because 10-11 do not seem to suggest a valid date. <br>
For example, `add buy milk from 7-11` 
The 7-11 here is a store and should not be seen as a date

File: src/main/java/seedu/taskell/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -52,89 +51,89 @@
...
+                startDate = today;
             }
-        } else {
-            return true;
         }
     }
 
-    private boolean isValidTime(ReadOnlyTask taskToEdit) {
+    private boolean ableToAdjustTime(ReadOnlyTask taskToEdit) {
+        TaskTime currentTime = TaskTime.getTimeNow();
```
Comment (damithc) [inline src/main/java/seedu/taskell/logic/commands/EditCommand.java:110]: You could have avoided a whole nesting level with this:

``` java
if (!taskToEdit.getTaskType().equals(Task.EVENT_TASK)) {
    return false;
}
```

File: src/main/java/seedu/taskell/logic/parser/Parser.java [hunk 2]
```diff
@@ -256,182 +257,179 @@ private Command prepareListPriority(String args) {
...
-                } else {
-                    return new IncorrectCommand(
-                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
-                }
+        } else if (parameters.equals(START_TIME)) {
+            if (canUpdateStartTime(newValue)) {
+                return true;
+            } else {
+                return false;
             }
```
Comment (damithc) [inline src/main/java/seedu/taskell/logic/parser/Parser.java:346]: `return canUpdateStartTime(newValue);` :-p

File: src/main/java/seedu/taskell/logic/parser/Parser.java [hunk 3]
```diff
@@ -256,182 +257,179 @@ private Command prepareListPriority(String args) {
...
+
+    private boolean canUpdatePriority(String newValue) {
+        taskComponentArray[Task.TASK_PRIORITY] = newValue;
+        if (TaskPriority.isValidPriority(taskComponentArray[Task.TASK_PRIORITY])) {
+            hasTaskComponentArray[Task.PRIORITY_COMPONENT] = true;
+            return true;
+        } else {
+            return false;
         }
+    }
```
Comment (damithc) [inline src/main/java/seedu/taskell/logic/parser/Parser.java:410]: These 4 methods seem to be very similar. Perhaps the similar code can be extracted into a more general method?

File: src/main/java/seedu/taskell/logic/parser/Parser.java [hunk 4]
```diff
@@ -441,259 +439,258 @@ private Command prepareAdd(String args) {
...
+                return new IncorrectCommand(ive.getMessage());
+            }
+        } else {
+            try {
+                return new AddCommand(Task.FLOATING_TASK, taskComponentArray, hasTaskComponentArray,
+                        getTagsFromArgs(taskComponentArray[Task.TAG]));
+            } catch (IllegalValueException ive) {
+                return new IncorrectCommand(ive.getMessage());
+            }
+        }
```
Comment (damithc) [inline src/main/java/seedu/taskell/logic/parser/Parser.java:481]: The two branches are very similar. Extract similar code into a general method?

File: src/main/java/seedu/taskell/model/task/EventTask.java [hunk 5]
```diff
@@ -14,53 +11,158 @@
...
+    }
+    
+    private void initialiseEndDate(TaskDate eventStartDate) {
+        if (!hasTaskComponentArray[Task.END_DATE_COMPONENT]) {
+            taskComponentArray[Task.END_DATE] = eventStartDate.toString();
+        }
+    }
+    
+    /**
+     * Determine end date in the week with respect to start date
```
Comment (damithc) [inline src/main/java/seedu/taskell/model/task/EventTask.java:133]: Determines

**Pull request merged**